src/backend/distributed/master/master_repair_shards.c

```diff
@@ -297,7 +296,8 @@ CopyShardPlacement(int64 shardId, char *sourceNodeName, int32 sourceNodePort,
 
 	HOLD_INTERRUPTS();
 
-	SendCommandListInSingleTransaction(targetNodeName, targetNodePort, ddlCommandList);
+	SendCommandListToWorkerInSingleTransaction(targetNodeName, targetNodePort, nodeUser,
```

Comment:
Yes, table owner would make sense. worker_transactions.c connects as the extension owner in order to make changes to pg_dist_ tables.

Creating the shard placement should not use 2PC. Otherwise, the new placement metadata might become visible before the shard placement becomes visible. Dropping the shard placement should use 2PC, otherwise the old shard placement might be dropped before the new placement metadata becomes visible.

Dropping the placement as the as extension owner should be ok, so you can use worker_transactions for that.